### PR TITLE
Added submission of nine box return data to backend

### PR DESCRIPTION
**Pull request by miranda-hawkes**
- [x] Merge after https://github.com/hmrc/submit-vat-return-frontend/pull/25 or Tom will tear his hair out (this will need rebasing)

### CODE REVIEW

File: it/base/BaseISpec.scala [hunk 1]
```diff
@@ -43,7 +44,8 @@ trait BaseISpec extends WordSpec with WireMockHelper with Matchers with
     "microservice.services.vat-returns.host" -> WireMockHelper.wireMockHost,
     "microservice.services.vat-returns.port" -> WireMockHelper.wireMockPort.toString,
     "microservice.services.auth.host" -> WireMockHelper.wireMockHost,
-    "microservice.services.auth.port" -> WireMockHelper.wireMockPort.toString
+    "microservice.services.auth.port" -> WireMockHelper.wireMockPort.toString,
+    "play.filters.csrf.header.bypassHeaders.Csrf-Token" -> "nocheck"
```
Comment (TomDudley92) [inline it/base/BaseISpec.scala]: this is already on line 39

File: test/controllers/ConfirmSubmissionControllerSpec.scala [hunk 2]
```diff
@@ -150,25 +152,112 @@ class ConfirmSubmissionControllerSpec extends BaseSpec with MockAuth with MockMa
...
+            Jsoup.parse(bodyOf(result)).title() shouldBe "Sorry, we are experiencing technical difficulties - 500"
+          }
+        }
+      }
+
+      "invalid session data exists" should {
+
+        val nineBoxData = Json.obj("box10" -> "why").toString()
+        lazy val result: Future[Result] = TestConfirmSubmissionController.submit("18AA")(fakeRequest.withSession(
+          "mtdNineBoxReturnData" -> nineBoxData,
```
Comment (TomDudley92) [inline test/controllers/ConfirmSubmissionControllerSpec.scala]: Can you use the SessionKey here please and the other examples within this file as well please?

File: test/controllers/ConfirmSubmissionControllerSpec.scala [hunk 3]
```diff
@@ -150,25 +152,112 @@ class ConfirmSubmissionControllerSpec extends BaseSpec with MockAuth with MockMa
...
+            mockVatReturnsService(Future.successful(Right(SubmissionSuccessModel("12345"))))
+            status(result) shouldBe Status.SEE_OTHER
+          }
+
+          s"redirect to ${controllers.routes.ConfirmationController.show()}" in {
+            redirectLocation(result) shouldBe Some(controllers.routes.ConfirmationController.show().url)
+          }
+
+          "remove form data from session" in {
+            result.session.get("mtdNineBoxReturnData") shouldBe None
```
Comment (TomDudley92) [inline test/controllers/ConfirmSubmissionControllerSpec.scala]: This test will always pass due to the way session data works with tests. Unless you explicitly add something to session within this test scope, nothing is kept in session. So it isn't really proving anything

File: test/controllers/ConfirmSubmissionControllerSpec.scala (hunk 2 above)
Comment (miranda-hawkes) [reply test/controllers/ConfirmSubmissionControllerSpec.scala]: I prefer typing it in so its almost like a test for the session key identifier being correct too (similar to having a separate message lookup in tests)

**Pull request merged**